Code/Tools/SceneAPI/SceneCore/Utilities/PatternMatcher.cpp

```diff
@@ -106,51 +122,75 @@ namespace AZ
                 case MatchApproach::PostFix:
-                {
-                    if (m_pattern.size() > nameLength)
+                    for (const auto& pattern : m_patterns)
                     {
-                        return false;
+                        // Perform a case-insensitive postfix match.
+                        if (AZ::StringFunc::EndsWith(name, pattern, caseSensitive))
+                        {
+                            return true;
+                        }
                     }
-                    size_t offset = nameLength - m_pattern.size();
-                    return AzFramework::StringFunc::Equal(name + offset, m_pattern.c_str());
-                }
+                    return false;
                 case MatchApproach::Regex:
                 {
-                    // Because PatternMatcher can get default constructed and serialized into directly, there's no good place
-                    // to initialize the regex matcher on construction, so we'll lazily initialize it here on first use.
-                    if (m_regexMatcher == nullptr)
+                    for (size_t index = 0; index < m_patterns.size(); index++)
                     {
-                        m_regexMatcher = AZStd::make_unique<AZStd::regex>(m_pattern, AZStd::regex::extended);
+                        // Because PatternMatcher can get default constructed and serialized into directly, there's no good place
+                        // to initialize the regex matchers on construction, so we'll lazily initialize it here on first use.
+
+                        if (m_regexMatchers.empty())
+                        {
+                            m_regexMatchers.resize(m_patterns.size());
+                        }
+
+                        if (m_regexMatchers[index] == nullptr)
+                        {
+                            m_regexMatchers[index] = AZStd::make_unique<AZStd::regex>(m_patterns[index], AZStd::regex::extended);
+                        }
+                        AZStd::cmatch match;
+                        if (AZStd::regex_match(name.begin(), name.end(), match, *(m_regexMatchers[index])))
+                        {
+                            return true;
+                        }
                     }
-                    AZStd::smatch match;
-                    return AZStd::regex_match(name, match, *m_regexMatcher);
+                    return false;
                 }
                 default:
                     AZ_Assert(false, "Unknown option '%i' for pattern matcher.", m_matcher);
                     return false;
                 }
             }
 
-            bool PatternMatcher::MatchesPattern(const AZStd::string& name) const
-            {
-                return MatchesPattern(name.c_str(), name.length());
-            }
-
             const AZStd::string& PatternMatcher::GetPattern() const
             {
-                return m_pattern;
+                return m_patterns[0];
```

Comment:
When a PatternMatcher that was created with a default constructor is used, this will perform an out of bounds reference.

This can return an `const char*, `AZStd::string_view`, `AZStd::string` or it can return a empty static `AZStd::string`.
```suggestion
                const static AZStd::string emptyString{};
                return !m_patterns.empty() ? m_patterns.front() : emptyString;
```